cvmfs/catalog.cc

```diff
@@ -408,6 +415,48 @@ void Catalog::DropDatabaseFileOwnership() {
 }
 
 
+Catalog::ExternalDataStatus Catalog::GetExternalDataUnlocked() const {
+  ExternalDataStatus status = \
+    static_cast<ExternalDataStatus>(atomic_read32(&external_data_status_));
+  if (status == kExternalUnknown) {
+    bool attr = false;
+    if (database().HasProperty("external_data")) {
+      attr = database().GetProperty<int>("external_data");
+      status = attr ? kExternalPresent : kExternalNone;
+    } else {
+      status = parent_ ? parent_->GetExternalDataLocked() : \
+                         kExternalUnspecified;
+    }
+    atomic_write32(&external_data_status_, status);
+  }
+  return status;
+}
+
+Catalog::ExternalDataStatus Catalog::GetExternalDataLocked() const {
```

Comment:
That's code duplication with `Catalog::GetExternalDataUnlocked`... Could that be DRY-ed up?